#!/usr/bin/env python
import roslib;roslib.load_manifest('cob_teleop_cob4')

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("button_deadman", int_t, 0, "Autogenerated parameter based on model.", 11, -100, 100)
gen.add("base_max_linear", double_t, 0, "Autogenerated parameter based on model.", 0.5, -100, 100)
gen.add("base_max_angular", double_t, 0, "Autogenerated parameter based on model.", 1.5, -100, 100)
gen.add("torso_max_angular", double_t, 0, "Autogenerated parameter based on model.", 0.2, -100, 100)
gen.add("head_max_angular", double_t, 0, "Autogenerated parameter based on model.", 0.3, -100, 100)
gen.add("sensor_ring_max_angular", double_t, 0, "Autogenerated parameter based on model.", 0.1, -100, 100)
gen.add("arm_joint_velocity_max", double_t, 0, "Autogenerated parameter based on model.", 0.3, -100, 100)
gen.add("arm_cartesian_max_linear", double_t, 0, "Autogenerated parameter based on model.", 0.1, -100, 100)
gen.add("arm_cartesian_max_angular", double_t, 0, "Autogenerated parameter based on model.", 0.1, -100, 100)
gen.add("gripper_max_velocity", double_t, 0, "Autogenerated parameter based on model.", 0.1, -100, 100)
gen.add("base_x", int_t, 0, "Autogenerated parameter based on model.", 1, -100, 100)
gen.add("base_y", int_t, 0, "Autogenerated parameter based on model.", 0, -100, 100)
gen.add("base_yaw", int_t, 0, "Autogenerated parameter based on model.", 2, -100, 100)
gen.add("arm_x", int_t, 0, "Autogenerated parameter based on model.", 0, -100, 100)
gen.add("arm_y", int_t, 0, "Autogenerated parameter based on model.", 1, -100, 100)
gen.add("arm_yaw", int_t, 0, "Autogenerated parameter based on model.", 2, -100, 100)
gen.add("arm_pitch_up", int_t, 0, "Autogenerated parameter based on model.", 4, -100, 100)
gen.add("arm_pitch_down", int_t, 0, "Autogenerated parameter based on model.", 6, -100, 100)
gen.add("arm_roll_right_and_ellbow", int_t, 0, "Autogenerated parameter based on model.", 5, -100, 100)
gen.add("arm_roll_left_and_ellbow", int_t, 0, "Autogenerated parameter based on model.", 7, -100, 100)
gen.add("arm_z_up", int_t, 0, "Autogenerated parameter based on model.", 12, -100, 100)
gen.add("arm_z_down", int_t, 0, "Autogenerated parameter based on model.", 14, -100, 100)
gen.add("gripper_open", int_t, 0, "Autogenerated parameter based on model.", 15, -100, 100)
gen.add("gripper_close", int_t, 0, "Autogenerated parameter based on model.", 13, -100, 100)
gen.add("arm_joint_up", int_t, 0, "Autogenerated parameter based on model.", 4, -100, 100)
gen.add("arm_joint_down", int_t, 0, "Autogenerated parameter based on model.", 6, -100, 100)
gen.add("arm_joint_left", int_t, 0, "Autogenerated parameter based on model.", 7, -100, 100)
gen.add("arm_joint_right", int_t, 0, "Autogenerated parameter based on model.", 5, -100, 100)
gen.add("arm_joint_12", int_t, 0, "Autogenerated parameter based on model.", 15, -100, 100)
gen.add("arm_joint_34", int_t, 0, "Autogenerated parameter based on model.", 14, -100, 100)
gen.add("arm_joint_56", int_t, 0, "Autogenerated parameter based on model.", 13, -100, 100)
gen.add("arm_joint_7_gripper", int_t, 0, "Autogenerated parameter based on model.", 12, -100, 100)
gen.add("axis_runfactor", int_t, 0, "Autogenerated parameter based on model.", 9, -100, 100)
gen.add("button_safety_override", int_t, 0, "Autogenerated parameter based on model.", 9, -100, 100)
gen.add("button_init_recover", int_t, 0, "Autogenerated parameter based on model.", 3, -100, 100)
gen.add("button_mode_switch", int_t, 0, "Autogenerated parameter based on model.", 0, -100, 100)
gen.add("torso_roll", int_t, 0, "Autogenerated parameter based on model.", 0, -100, 100)
gen.add("torso_pitch", int_t, 0, "Autogenerated parameter based on model.", 1, -100, 100)
gen.add("torso_yaw_left", int_t, 0, "Autogenerated parameter based on model.", 15, -100, 100)
gen.add("torso_yaw_right", int_t, 0, "Autogenerated parameter based on model.", 13, -100, 100)
gen.add("sensorring_yaw_left", int_t, 0, "Autogenerated parameter based on model.", 4, -100, 100)
gen.add("sensorring_yaw_right", int_t, 0, "Autogenerated parameter based on model.", 6, -100, 100)
gen.add("head_roll", int_t, 0, "Autogenerated parameter based on model.", 2, -100, 100)
gen.add("head_pitch", int_t, 0, "Autogenerated parameter based on model.", 3, -100, 100)
gen.add("head_yaw_left", int_t, 0, "Autogenerated parameter based on model.", 7, -100, 100)
gen.add("head_yaw_right", int_t, 0, "Autogenerated parameter based on model.", 5, -100, 100)
gen.add("head_home", int_t, 0, "Autogenerated parameter based on model.", 4, -100, 100)
gen.add("arm_left_home", int_t, 0, "Autogenerated parameter based on model.", 7, -100, 100)
gen.add("arm_right_home", int_t, 0, "Autogenerated parameter based on model.", 5, -100, 100)
gen.add("torso_home", int_t, 0, "Autogenerated parameter based on model.", 6, -100, 100)
gen.add("sensorring_home", int_t, 0, "Autogenerated parameter based on model.", 12, -100, 100)
gen.add("gripper_left_home", int_t, 0, "Autogenerated parameter based on model.", 15, -100, 100)
gen.add("gripper_right_home", int_t, 0, "Autogenerated parameter based on model.", 13, -100, 100)
gen.add("base_home", int_t, 0, "Autogenerated parameter based on model.", 14, -100, 100)
gen.add("home_time", double_t, 0, "Autogenerated parameter based on model.", 5.0, -100, 100)
gen.add("stop_time", double_t, 0, "Autogenerated parameter based on model.", 0.8, -100, 100)
gen.add("gripper_1", int_t, 0, "Autogenerated parameter based on model.", 3, -100, 100)
gen.add("gripper_2", int_t, 0, "Autogenerated parameter based on model.", 4, -100, 100)
gen.add("gripper_max_angular", double_t, 0, "Autogenerated parameter based on model.", 0.2, -100, 100)

exit(gen.generate("cob_teleop_cob4", "cob_teleop_cob4", "cob_teleop_cob4"))
